src/super_gradients/common/code_analysis/breaking_change.py

```diff
@@ -0,0 +1,268 @@
+
+        if function_name in current_functions_signatures:
+            current_function_signature = current_functions_signatures[function_name]
+
+            # ParameterRemoved
+            for source_param in source_function_signature.params.all:
+                if source_param not in current_function_signature.params.all:
+                    breaking_changes.params_removed.append(
+                        ParameterRemoved(
+                            function_name=function_name,
+                            parameter_name=source_param,
+                            line_num=current_function_signature.line_num,
+                        )
+                    )
+
+            # RequiredParameterAdded
+            for current_param in current_function_signature.params.required:
+                if current_param not in source_function_signature.params.required:
+                    breaking_changes.required_params_added.append(
+                        RequiredParameterAdded(
+                            function_name=function_name,
+                            parameter_name=current_param,
+                            line_num=current_function_signature.line_num,
+                        )
+                    )
+
+        else:
+            # FunctionRemoved
+            breaking_changes.functions_removed.append(
+                FunctionRemoved(
+                    function_name=function_name,
+                    line_num=source_function_signature.line_num,
+                )
+            )
+
+    # Check import ONLY if __init__ file.
+    if module_path.endswith("__init__.py"):
+        source_imports = parse_imports(code=source_code)
+        current_imports = parse_imports(code=current_code)
+        breaking_changes.imports_removed = [
+            ImportRemoved(import_name=source_import, line_num=0) for source_import in source_imports if source_import not in current_imports
+        ]
+    return breaking_changes
+
+
+def analyze_breaking_changes(verbose: bool = 1) -> List[Dict[str, Union[str, List]]]:
+    """Analyze changes between the current branch (HEAD) and the master branch.
+    :param verbose: If True, print the summary of breaking changes in a nicely formatted way
+    :return:        List of changes, where each change is a dictionary listing each type of change for each module.
+    """
+    # GitHelper requires `git` library which should NOT be required for the other functions
+    from super_gradients.common.code_analysis.git_utils import GitHelper
+
+    root_dir = os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(__file__)))))
+    git_explorer = GitHelper(git_path=root_dir)
+
+    summary = ""
+    breaking_changes_list = []
+    for module_path in git_explorer.diff_files(source_branch="master", current_branch="HEAD"):
+
+        master_code = git_explorer.load_branch_file(branch="master", file_path=module_path)
```

Comment:
will this catch a file with a changed name?
i.e. if the file was 
myClass.py
and it is now 
myNewClass.py

Reply:
The way it is currently detected is that when a file was renamed, the old file is seen as deleted and a new one as created.
Therefore, any function/class that was in that renamed file will be considered as deleted.

It's not as clean as simply having "module renamed", but eventually we get what we want; old functions/classes are not available in the module they used to.
Do you think we should catch it explicitly with a `ModuleRenamed` case?